Kernel/Arch/aarch64/PSCI.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2025, the SerenityOS developers
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/Arch/aarch64/PSCI.h>
+#include <Kernel/Firmware/DeviceTree/DeviceTree.h>
+
+namespace Kernel::PSCI {
+
+#define PSCI_FUNCTION_BASE 0x84000000
+
+#define PSCI_SYSTEM_OFF ((PSCI_FUNCTION_BASE) + 8)
+#define PSCI_SYSTEM_RESET ((PSCI_FUNCTION_BASE) + 9)
+
+static inline FlatPtr aarch64_hvc(FlatPtr arg0, FlatPtr arg1, FlatPtr arg2, FlatPtr arg3, FlatPtr arg4, FlatPtr arg5, FlatPtr arg6, FlatPtr arg7)
```

Comment:
I wouldn't use such a generic name for this function. This specifically assumes the SMCCC/AAPCS64 calling convention.

Maybe just name it `PSCI::call` and make it take only 4 arguments `(Function function, FlatPtr arg0, FlatPtr arg1, FlatPtr arg2)`, since PSCI only supports up to 4 arguments?

Also, referencing "5.2.1 Register usage in arguments and return value" would be nice here.

Reply:
Also, the `inline` should be unnecessary.